mindsdb/interfaces/agents/agents_controller.py

```diff
@@ -180,8 +179,30 @@ def add_agent(
         if agent is not None:
             raise ValueError(f"Agent with name already exists: {name}")
 
+        # Get default LLM parameters from config if they exist
+        default_llm_params = config.get("default_llm", {})
+
+        # If model_name is not provided but available in default_llm config, use it
+        if model_name is None and default_llm_params and "model_name" in default_llm_params:
+            model_name = default_llm_params.get("model_name")
+
+        # If provider is not provided but available in default_llm config, use it
+        if provider is None and default_llm_params and "provider" in default_llm_params:
+            provider = default_llm_params.get("provider")
+
+        # Check if model_name is still None after trying to get it from default_llm
+        if model_name is None:
+            raise ValueError("Model name must be provided either directly or through default_llm configuration")
+
         _, provider = self.check_model_provider(model_name, provider)
 
+        # Initialize params with a copy to avoid modifying the input
+        params = (params or {}).copy()
```

Comment:
Do we need this line? The copy is just overriding the original variable. It should either be changed to a different variable name or just removed.